### PR TITLE
Bug 1874522 - Repalce coil-kt with AndroidSVG

### PR DESCRIPTION
**Pull request by jackyzy823**
1. Use AndriodSVG directly
    It is safe to use AndroidSVG , because it is the dependency of coil-svg

    Since android-components has its own cache mechanism , using coil-kt is just bloating.

2. Catch OOM error just like https://github.com/mozilla-mobile/firefox-android/blob/76213dc84f49a0400e009ea06e3d09fd3ad68ded/android-components/components/support/images/src/main/java/mozilla/components/support/images/decoder/AndroidImageDecoder.kt#L36
3. Set target bitmap size according to desiredSize to avoid svg too large which could also cause OOM

The code is **largely borrowed** from  https://github.com/coil-kt/coil/blob/2.4.0/coil-svg/src/main/java/coil/decode/SvgDecoder.kt with some fixed options.

( **However  I'm not sure if this patch will really fix the bug. and also i'm not sure if the bug was caused by using coil-svg**)

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
5. Click on `View task in Taskcluster` in the new `DETAILS` section.
6. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1879116

### CODE REVIEW

**Comment by Roukanken42** [comment]
As I have managed to reproduce this bug consistently on any other build and both my android devices, I also tested build from this PR's CI artefacts and it fixes both ways I could reproduce the issue.

**Comment by t-p-white** [comment]
@Roukanken42 🙌 Can I confirm [your comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1874522#c29) on Bugzilla that using the [patch here](https://bugzilla.mozilla.org/show_bug.cgi?id=1874522#c28) also resolves the bug?

**Comment by Roukanken42** [comment]
@t-p-white Yes, the patch also resolves the bug.

**Comment by t-p-white** [comment]
@Roukanken42 perfect, thank you for confirming 🙏 

@jackyzy823, nice work 👌  please see my latest comment on Bugzilla regarding this particular change - but essentially I think we should use your original patch for `SvgIconDecoder` for this bug and create a follow on if a library update is required 🙂

**Review by t-p-white — APPROVED** [review]

**Comment by t-p-white** [comment]
Apologies, I got this PR mixed up with https://github.com/mozilla-mobile/firefox-android/pull/5443

**Comment by gabrielluong** [comment]
@t-p-white I am wondering if you will have any capacity to review this?

**Comment by t-p-white** [comment]
Thanks for the ping @gabrielluong. Yes, it's on my radar todo 🙂

File: android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/SvgIconDecoder.kt [hunk 1]
```diff
@@ -6,55 +6,67 @@ package mozilla.components.browser.icons.decoder
...
+import androidx.core.graphics.createBitmap
+import com.caverock.androidsvg.SVG
+import com.caverock.androidsvg.SVGParseException
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.images.DesiredSize
 import mozilla.components.support.images.decoder.ImageDecoder
 
 /**
- * [ImageDecoder] that will use the Coil library [SvgDecoder.Factory] in order to decode the byte data.
+ * [ImageDecoder] that will use the AndroidSVG in order to decode the byte data.
```
Comment (t-p-white) [inline android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/SvgIconDecoder.kt:20]: >The code is largely borrowed from https://github.com/coil-kt/coil/blob/2.4.0/coil-svg/src/main/java/coil/decode/SvgDecoder.kt with some fixed options.

Can you add a comment like this to the fun description please?

File: android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/SvgIconDecoder.kt [hunk 2]
```diff
@@ -6,55 +6,67 @@ package mozilla.components.browser.icons.decoder
...
+            } else {
+                svgWidth = svg.documentWidth
+                svgHeight = svg.documentHeight
+            }
+
+            var bitmapWidth = desiredSize.targetSize
+            var bitmapHeight = desiredSize.targetSize
+
+            if (svgWidth > 0 && svgHeight > 0) {
+                val widthPercent = bitmapWidth / svgWidth.toDouble()
```
Comment (t-p-white) [inline android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/SvgIconDecoder.kt:45]: Nice 👌 
Perhaps a comment on this block will be useful for future maintainers? Maybe something like "Scale the bitmap to SVG maintaining the aspect ratio"

**Review by t-p-white — APPROVED** [review]
Thanks @jackyzy823!

**Pull request merged**